### PR TITLE
Add endpoint to kill an AWS worker

### PR DESCRIPTION
**Pull request by helfi92**

### CODE REVIEW

File: src/api-v1.js [hunk 1]
```diff
@@ -414,6 +414,45 @@ api.declare({
 });
 
 api.declare({
+  method:     'delete',
+  route:      '/killWorker',
+  // actions with context=worker will a query of the form ?provisionerId=...&workerType=...&workerGroup=...&workerId=...
+  // https://docs.taskcluster.net/reference/platform/taskcluster-queue/docs/actions#context
+  query: {
```
Comment (jhford) [inline src/api-v1.js]: Query strings are only for continuation token and limit, so these inputs shouldn't be provided here.

Please use route parameters instead.

File: src/api-v1.js [hunk 2]
```diff
@@ -414,6 +414,45 @@ api.declare({
 });
 
 api.declare({
+  method:     'delete',
+  route:      '/killWorker',
```
Comment (jhford) [inline src/api-v1.js]: this is not a REST endpoint.  This is also not the correct service to have this endpoint.

The correct URL to use is:

```
https://ec2-manager.taskcluster.net/v1/region/{WORKERGROUP}/instance/{WORKERID}
```

File: src/api-v1.js [hunk 3]
```diff
@@ -414,6 +414,45 @@ api.declare({
...
+  // actions with context=worker will a query of the form ?provisionerId=...&workerType=...&workerGroup=...&workerId=...
+  // https://docs.taskcluster.net/reference/platform/taskcluster-queue/docs/actions#context
+  query: {
+    provisionerId: /./,
+    workerType: /./,
+    workerGroup: /./,
+    workerId: /./,
+  },
+  name:       'killWorker',
+  scopes: [['aws-provisioner:kill-worker:<provisionerId>/<workerType>/<workerGroup>/<workerId>']],
```
Comment (jhford) [inline src/api-v1.js]: We should be using the `aws-provisioner:manage-worker-type:<workerType>` scope.

File: src/api-v1.js [hunk 4]
```diff
@@ -414,6 +414,45 @@ api.declare({
...
+  input:      undefined,
+  output:     undefined,
+  title:      'Kill a Worker',
+  stability:  API.stability.experimental,
+  description: [
+    'Kill an AWS worker by supplying a query with its region (workerGroup) and instanceId (workerId).',
+  ].join('\n'),
+}, async function(req, res) {
+  const {workerGroup, workerId} = req.query;
```
Comment (jhford) [inline src/api-v1.js]: Please switch this to route parameters.

File: src/api-v1.js [hunk 5]
```diff
@@ -414,6 +414,45 @@ api.declare({
...
+
+  if (!workerGroup || !workerId) {
+    res.status(400).end();
```
Comment (jhford) [inline src/api-v1.js]: please use `res.reportError()`

**Comment by helfi92** [comment]
If we're taking the approach of path parameters instead of query, we won't be needing this api endpoint anymore. Closing this for now.

**Pull request closed without merging**